minor fix to fused adam meta registration

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99901
* #99900
* #99899
* #99898
* #99374
* #99489
* __->__ #99436
* #99897

This PR fixes the registration by adding `max_exp_avg_sqs` to the
output shape list too, and fix some type check issue

wanchaol committed:
Add meta registration for aten._fused_adam.default

This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

mrshenli commented on torch/distributed/_spmd/api.py:
```diff
@@ -268,7 +268,7 @@ def _foreach_addcop_scalar_decomp(op, self, tensor1, tensor2, scalar=1):
 
 
 def _fused_adam_decomp(
-    self,
```
just for context, using self here is to comply with aten signature. 


```
'aten::_fused_adam(Tensor[] self, Tensor[] grads, Tensor[] exp_avgs, Tensor[] exp_avg_sqs, Tensor[] max_exp_avg_sqs, Tensor[] state_steps, *, float lr, float beta1, float beta2, float weight_decay, float eps, bool amsgrad, bool maximize, Tensor? grad_scale=None, Tensor? found_inf=None) -> (Tensor[] self_out, Tensor[] grads_out, Tensor[] exp_avgs_out, Tensor[] exp_avg_sqs_out, Tensor[] max_exp_avg_sqs_out)'
```

mrshenli commented on torch/_meta_registrations.py:
```diff
@@ -1398,13 +1398,50 @@ def meta__fused_adam_(
         )
 
 
+@register_meta([aten._fused_adam.default])
```
does this address the issue you encountered?

wanchaol replied:
Yep! resolved the most part, thanks for the suggestion! I think there're still some issues with regard to fake tensor fallback "val" set, documented in https://github.com/pytorch/pytorch/issues/99356

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "Add meta registration for aten._fused_adam.default"


This PR adds meta registration for the out of place fused_adam call

[ghstack-poisoned]

wanchaol committed:
Update on "minor fix to fused adam meta registration"


This PR fixes the registration by adding `max_exp_avg_sqs` to the
output shape list too, and fix some type check issue

[ghstack-poisoned]

wanchaol committed:
Update on "minor fix to fused adam meta registration"


This PR fixes the registration by adding `max_exp_avg_sqs` to the
output shape list too, and fix some type check issue

[ghstack-poisoned]
